langchain4j-pinecone/src/test/java/dev/langchain4j/store/embedding/pinecone/PineconeEmbeddingStoreWithMetadataIT.java

```diff
@@ -0,0 +1,130 @@
+package dev.langchain4j.store.embedding.pinecone;
+
+import com.google.protobuf.Value;
+import dev.langchain4j.data.document.Metadata;
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.model.embedding.AllMiniLmL6V2QuantizedEmbeddingModel;
+import dev.langchain4j.model.embedding.EmbeddingModel;
+import dev.langchain4j.store.embedding.EmbeddingMatch;
+import dev.langchain4j.store.embedding.EmbeddingSearchRequest;
+import dev.langchain4j.store.embedding.EmbeddingStore;
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static dev.langchain4j.internal.Utils.randomUUID;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.data.Percentage.withPercentage;
+
+@EnabledIfEnvironmentVariable(named = "PINECONE_API_KEY", matches = ".+")
+class PineconeEmbeddingStoreWithMetadataIT {
+
+    EmbeddingModel embeddingModel = new AllMiniLmL6V2QuantizedEmbeddingModel();
+
+    EmbeddingStore<TextSegment> embeddingStore = PineconeEmbeddingStore.builder()
+            .apiKey(System.getenv("PINECONE_API_KEY"))
+            .cloud("AWS")
+            .region("us-east-1")
+            .index("test")
+            .nameSpace(randomUUID())
+            .dimension(embeddingModel.dimension())
+            .metadataTypeMap(toMetadataTypeMap())
+            .createIndex(true)
+            .build();
+
+    @Test
+    void should_add_embedding_with_segment_with_metadata() {
```

Comment:
Does this test exists because the one from `EmbeddingStoreIT` has more types than Pinecone supports? Can we somehow map all LC4j supported types to Pinecone types? E.g. integer, long, float (java) -> double (Pinecone), etc?

Reply:
Good idea I think. But the question is how to infer its Java type when searching.

Reply:
Currently the IT passed. But in metadata of searching result, all value's Java type are changed to double

Reply:
Do you mean in the `Metadata.toMap()` all ints/longs/floats are doubles in the search results? I guess it is ok. `Metadata.getXXX` methods will do the conversion.

Reply:
OK! Now all `langchain4j` supported types can map to `Pinecone` types.